core/src/main/java/hudson/ProxyConfiguration.java

```diff
@@ -157,15 +162,15 @@ public String getUserName() {
         return userName;
     }
 
-//    This method is public, if it was public only for jelly, then should make it private (or inline contents)
-//    Have left public, as can't tell if anyone else is using from plugins
-    /**
-     * @return the password in plain text
-     */
-    public String getPassword() {
-        return Secret.toString(secretPassword);
+    public Secret getPassword() {
```

Comment:
I found some usages of the API in some proprietary plugins. My suggestion would be to retain API if possible, but @daniel-beck suggests we go forward to ensure we have the safe API.

```suggestion
    @Deprecated
    public String getPassword() { 
       return Secret.toString(secretPassword);
    }
    
    public Secret getSecretPassword() {
```